libs/components/core/testing/src/shared/input-harness.ts

```diff
@@ -0,0 +1,60 @@
+import { ComponentHarness } from '@angular/cdk/testing';
+
+/**
+ * Harness used to interact with native input elements in tests.
+ * @internal
+ */
+export class SkyInputHarness extends ComponentHarness {
+  /**
+   * Blurs the input.
+   */
+  public async blur(): Promise<void> {
+    return (await this.host()).blur();
+  }
+
+  /**
+   * Clears the input value.
+   */
+  public async clear(): Promise<void> {
+    return (await this.host()).clear();
+  }
+
+  /**
+   * Enters text into the input.
+   */
+  public async enterText(value: string): Promise<void> {
```

Comment:
would this account for native inputs like `select` that dont use keys? looking at angular material's version two things stand out 
- their description calls out that they are simulating key presses. can we change the jsdocs to say something like `Simulates key presses to set a value to the input`
- can we check if the input has been updated to the new value and if not, call the `setInputValue` function to make sure that we dont change the value twice, but also account for non key press inputs
``` 
if(await this.getValue() !== value) {
 await el.setInputValue(value)
}
```

Reply:
For your second point, I think it's better to let the consumer of the test call `setValue()` as many times as they want even if it's the same value since the end user could also do that (we're simulating human interactions). Also, the consumer of the input harness might want to call `setValue()` multiple times because they're testing how their input directive handles duplicate value assignments.

For example, you might see a consumer create a test like this:
```
const updateSpy = spyOn(some-service-call);
myInputHarness.setValue('foo');
myInputHarness.setValue('foo');
myInputHarness.setValue('foo');
expect(updateSpy).toHaveBeenCalledTimes(1);
```